docs/style.rst

```diff
@@ -36,51 +36,154 @@ We like to use `>&` for file descriptors (numbers), and `&>` for filenames
 
      exec 3>&-
 
-     exec 3&>- # WRONG style. This is dealing with descriptors
+     exec 3&>- # Wrong style. This is dealing with descriptors
 
 * Redirecting to a file
 
   .. code-block:: bash
 
      run_some_command &> /dev/null # Spaces added around &>
 
-     run_some_command >& /dev/null # WRONG style. This is dealing with filenames
+     run_some_command >& /dev/null # Wrong style. This is dealing with filenames
 
-* Always quote variables
+* Always quote variables and complex strings
+
+  There are many conditions when you need to use quotes, and very few when you need not to, and many when quotes seem optional. Always using quotes vastly reduces the chance of errors from unexpected (and erroneous) input and simplifies remembering when you need and don't need quotes
 
   .. code-block:: bash
 
-    avar=foo
-    echo "${avar}"
+     bvar=${avar}   # Wrong style. Always add quotes
+     bvar="${avar}" # Right
+     cvar=foo bar   # Wrong style and syntax. This is a complex string, and requires quotes/
+                    # This actually executes a command called bar. Instead, make it clear
+     cvar="foo" bar # Right
+     cvar="foo bar" # Right
+     echo ${avar}   # Wrong style. Whitespace is not replicated the same way without quotes
+     echo "${avar}"
+
+     # There are a few reason to not quote a variable. Be sure to add a "# noquotes"
+     # comment so code reviewers knows these are intended:
+
+     # One is an empty variable that should not be treated as an empty string
+     # DRYRUN/optional_flag could be "echo"/"-stuff" or empty string
+     ${DRYRUN} some command # noquotes
+     some_command ${optional_flag} foo bar # noquotes
+     # The preferred way is to use arrays, if it is not overly cumbersome.
+     ${DRYRUN[@]+"${DRYRUN[@]}"} some command
+     some_command ${optional_flag[@]+"${optional_flag[@]}"} foo bar
+     # Since arrays cannot be exported, this is often not viable.
+
+     # Another reason quotes must not be used, is when splitting up a string into an array
+     foo="aa:bb:cc:dd"
+     IFS=":"
+     # This is actively splitting apart a string, and must not be in quotes
+     bar=(${foo}) # noquotes
```

Comment:
Should this be `bar=(${foo[@]})`

Reply:
No, foo is a string, not an array. bar is the array